edi_oca/models/edi_backend.py

```diff
@@ -291,411 +148,71 @@ def _validate_data(self, exchange_record, value=None, **kw):
-        domain = [
-            ("type_id.direction", "=", "output"),
-            ("backend_id", "=", self.id),
-            ("edi_exchange_state", "in", states),
-        ]
-        if record_ids:
-            domain.append(("id", "in", record_ids))
-        return domain
+        return super()._exchange_send(exchange_record)
 
     def _exchange_output_check_state(self, exchange_record):
         component = self._get_component(exchange_record, "check")
         if component:
             return component.check()
-        raise NotImplementedError("No handler for `_exchange_output_check_state`")
-
-    def _exchange_process_check(self, exchange_record):
-        if not exchange_record.direction == "input":
-            raise exceptions.UserError(
-                _("Record ID=%d is not meant to be processed") % exchange_record.id
-            )
-        if (
-            not exchange_record.exchange_file
-            and not exchange_record.type_id.allow_empty_files_on_receive
-        ):
-            raise exceptions.UserError(
-                _("Record ID=%d has no file to process!") % exchange_record.id
-            )
-        return exchange_record.edi_exchange_state in [
-            "input_received",
-            "input_processed_error",
-        ]
-
-    def exchange_process(self, exchange_record):
-        """Process an incoming document."""
-        self.ensure_one()
-        exchange_record.ensure_one()
-        # In case already processed: skip processing and check the state
-        check = self._exchange_process_check(exchange_record)
-        if not check:
-            return "Nothing to do. Likely already processed."
-        old_state = state = exchange_record.edi_exchange_state
-        error = False
-        message = None
-        try:
-            res = self._exchange_process(exchange_record)
-        except self._swallable_exceptions():
-            if self.env.context.get("_edi_process_break_on_error"):
-                raise
-            error = _get_exception_msg()
-            state = "input_processed_error"
-            res = f"Error: {error}"
-        else:
-            error = None
-            state = "input_processed"
-        finally:
-            exchange_record.write(
-                {
-                    "edi_exchange_state": state,
-                    "exchange_error": error,
-                    # FIXME: this should come from _compute_exchanged_on
-                    # but somehow it's failing in send tests (in record tests it works).
-                    "exchanged_on": fields.Datetime.now(),
-                }
-            )
-            if (
-                state == "input_processed_error"
-                and old_state != "input_processed_error"
-            ):
-                exchange_record._notify_error("process_ko")
-            elif state == "input_processed":
-                exchange_record._notify_done()
-        exchange_record.notify_action_complete("process", message=message)
```

Comment:
@acsonefho can you please restore the calls to `notify_action_complete`? This is a common interface to trigger stuff when something happens on a record. The only thing to re-implement would be `_trigger_edi_event`. 
There, you can leave it empty in edi_core and edi_oca will only override this to trigger events.

Reply:
@acsonefho are you going to move on soon w/ this?
We have pipeline of migration PR that are kept on hold for this.
Would be nice if we can integrate this right away, otherwise it will have to wait.
Just let me know. Thanks!